recipes/imagemagick/all/conanfile.py

```diff
@@ -61,375 +75,321 @@ class ImageMagicConan(ConanFile):
 
     def source(self):
-        tools.get(
-            **self.conan_data["sources"][self.version]["source"],
-            destination=self._source_subfolder,
-            strip_root=True
-        )
-
-        if self._is_msvc:
-            visualmagick_version = list(
-                self.conan_data["sources"][self.version]["visualmagick"].keys()
-            )[0]
-            tools.get(
-                **self.conan_data["sources"][self.version]["visualmagick"][
-                    visualmagick_version
-                ],
-                destination="VisualMagick",
-                strip_root=True
-            )
+        files.get(self,
+                  **self.conan_data["sources"][self.version],
+                  strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        env = tc.environment()
+
+        if microsoft.is_msvc(self):
+            # FIXME: it seems that PKG_CONFIG_PATH is added as a unix style path
+            env.unset("PKG_CONFIG_PATH")
+            env.define("PKG_CONFIG_PATH", self.generators_folder)
+
+            # FIXME: otherwise configure reports "ld" as its linker
+            env.define("LD", "link")
+            # use "compile" script as a compiler driver for cl, since autotools
+            # doesn't work well with cl
+            compile_script = microsoft.unix_path(
+                self, os.path.join(self.source_folder, "config", "compile"))
+            env.define("CC", f'{compile_script} cl.exe -nologo')
+            env.define("CXX", f"{compile_script} cl.exe -nologo")
+
+            env.append("CXXFLAGS", "-FS")
+            env.append("CFLAGS", "-FS")
+
+        tc.generate(env)
+
+        # FIXME: without this, AutotoolsDeps generates unix style path
+        self.win_bash = False
+        td = AutotoolsDeps(self)
+
+        # AutotoolsDeps uses /LIBPATH: which is a linker argument that
+        # will be pased incorrectly to the msvc compiler by autotools
+        # this replaces /LIBPATH: by  -L, which the "compile" script will
+        # fix for us, and replaces /I by -I, which works with the compile script
+        env = td.environment
+        if microsoft.is_msvc(self):
+            ldflags = env.vars(self)["LDFLAGS"].replace("/LIBPATH:", "-L").replace("\\", "/")
+            cppflags = env.vars(self)["CPPFLAGS"].replace("/I", "-I").replace("\\", "/")
+            env.define("LDFLAGS", ldflags)
+            env.define("CPPFLAGS", cppflags)
+
+        # AutotoolsDeps adds all dependencies in the libs variable.
+        # all these libs make `libtool` do the extra work of finding where they
+        # are, which seems to fail on Windows (and macOS as well, due to
+        # different reasons). this variable is not needed anyways, since
+        # autotools will link against the required libraries by itself
+        env.unset("LIBS")
+
+        td.generate()
+
+        self.win_bash = True
+
+        pd = PkgConfigDeps(self)
+        pd.generate()
```

Comment:
```suggestion
        pd.generate()
        env = VirtualBuildEnv(self)
        env.generate()
        if not cross_building(self):
            env = VirtualRunEnv(self)
            env.generate(scope="build")
```
Inject requires env vars in build scope. it's required in case of native build when there is AutotoolsDeps & at least one dependency which might be shared, because configure tries to run a test executable